packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -610,17 +557,18 @@ export abstract class BucketBase extends Resource implements IBucket {
    */
   protected abstract disallowPublicAccess?: boolean;
 
-  private notifications?: BucketNotifications;
+  protected factories: BucketFactories;
 
-  protected notificationsHandlerRole?: iam.IRole;
+  public notificationsHandlerRole?: iam.IRole;
 
-  protected notificationsSkipDestinationValidation?: boolean;
+  public notificationsSkipDestinationValidation?: boolean;
```

Comment:
Making these attributes public because `BucketFactories` needs to access them to create a `BucketNotifications` instance.

Reply:
So this is a `prop` that is only used in the factory, or rather the BucketNotifications class. I wonder if this could be kept private and passed into the factory, rather than being a reflection.

<img width="782" height="202" alt="image" src="https://github.com/user-attachments/assets/f1060d15-88e9-4355-8f19-8bf338036de7" />

Although since it's `protected` and not `readonly` today, it's effectively already public today...

Reply:
Yes, I just realized this is incorrect. I'm fixing it.